src/model_metric_reporter.hpp

```diff
@@ -0,0 +1,112 @@
+//*****************************************************************************
+// Copyright 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "execution_context.hpp"
+#include "metric.hpp"
+#include "modelversion.hpp"
+
+namespace ovms {
+
+class MetricRegistry;
+
+class ModelMetricReporter {
+    MetricRegistry* registry;
+
+public:
+    ModelMetricReporter(MetricRegistry* registry, const std::string& modelName, model_version_t modelVersion);
+
+    // TFS
+    std::shared_ptr<MetricCounter> requestSuccessGrpcPredict;
```

Comment:
Pretty minor but for developing macros It would be easier to have convention:
metricName(Success|Fail)

Reply:
Macros were not added, inline methods instead. https://github.com/openvinotoolkit/model_server/pull/1389